sympy/functions/special/delta_functions.py

```diff
@@ -562,7 +557,8 @@ class is about to be instantiated and it returns either some simplified
         elif arg.is_extended_positive:
             return S.One
         elif arg.is_zero:
-            return H0
+            if H0 is not nan:
+                return H0
```

Comment:
Do we not want to just return the `nan`?

Reply:
I would prefer that, actually, but then the information is lost that the nan comes from Heaviside's undefined step value. I thought a lot about this and concluded that returning nan must be OK; if anyone wants to maintain the info that the undefined value of Heaviside plays a role in some computation, they can call the function with Heaviside(x,Theta_0).

I'll change this.